src/Compilers/CSharp/Portable/Symbols/LocalSymbol.cs

```diff
@@ -387,6 +387,13 @@ protected sealed override ISymbol CreateISymbol()
             return new PublicModel.LocalSymbol(this);
         }
 
+        /// <summary>
+        /// Allows to access an underlying local symbol. If no underlying symbol exists - the same symbol is returned.
+        /// A typical use case is type substitution or a substituted containing-symbol (e.g. for the new extension mechanism).
+        /// Such substitutions are typically realized via <see cref="TypeSubstitutedLocalSymbol"/>
+        /// </summary>
+        internal virtual LocalSymbol UnderlyingLocalSymbol => this;
```

Comment:
>UnderlyingLocalSymbol [](http://example.com/codeflow?start=37&length=21)

I meant to suggest to add this API only to ```TypeSubstitutedLocalSymbol```. I guess we can add it to the base, but we prefer to not add virtual APIs because default implementations are bug prone in the long run. We prefer to add abstract APIs instead, implementing them in leaves. Therefore, sticking with an API in ```TypeSubstitutedLocalSymbol``` might be simpler.